### PR TITLE
planner: move logic of plan cache stmt checking to plan cache backend 

### PR DESCRIPTION
**Pull request by qw4990**
### What problem does this PR solve?

Issue Number: ref #36598

Problem Summary: planner: move logic of plan cache stmt checking to plan cache backend 

### What is changed and how it works?

No logical change, just move logic of plan cache stmt checking to plan cache backend 

### Check List

Tests 

- [ ] Unit test
- [x] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: planner/core/common_plans.go [hunk 1]
```diff
@@ -211,77 +206,6 @@ func isGetVarBinaryLiteral(sctx sessionctx.Context, expr expression.Expression)
...
-		prepared.CachedPlan = nil
-		vars.LastUpdateTime4PC = expiredTimeStamp4PC
-	}
-	plan, names, err := GetPlanFromSessionPlanCache(ctx, sctx, is, preparedObj, e.Params)
-	if err != nil {
-		return err
-	}
-	e.Plan = plan
-	e.names = names
-	e.Stmt = prepared.Stmt
```
Comment (Reminiscent) [inline planner/core/common_plans.go:281]: it seems we are not set this part in the new version?

Comment (qw4990) [reply planner/core/common_plans.go:281]: Yes, in the new version this part is useless.

**Review by Reminiscent — APPROVED** [review]

**Review by Yisaer — APPROVED** [review]

**Comment by hawkingrei** [comment]
/hold 

@qw4990 Please fix the test.

**Review by Reminiscent — APPROVED** [review]

**Review by Reminiscent — APPROVED** [review]

**Pull request merged**